### ISSUE TITLE
Health check not working

### ISSUE THREAD

**Issue by ryncsn** [tootsuite/mastodon#16242]
### Expected behaviour

The default health should just work, in docker-compose file:
```
    healthcheck:
      test: ["CMD-SHELL", "wget -q --spider --proxy=off localhost:3000/health || exit 1"]
```

Or in helm chart:
```
          livenessProbe:
            httpGet:
              path: /health
              port: http
          readinessProbe:
            httpGet:
              path: /health
              port: http
```

### Actual behaviour

Health check always return false.

### Steps to reproduce the problem

Spawn a mastodon container with image built with latest git repo, health check will always fail.

### Specifications

`git bisect` indicating following commit introduced this issue:

```
commit 83230234643bb53ba563e42d73fb91a0dcfbff64 (refs/bisect/bad)
Author: Takeshi Umeda <noel.yoshiba@gmail.com>
Date:   Thu Apr 22 00:45:58 2021 +0900

    Add guard against DNS rebinding attacks (#16087)
    
    * Add guard against DNS rebinding attacks
    
    * Fix not to apply to test environment

```

**Comment by shleeable** [comment]
Is this issue resolved by #16243?

**Comment by ClearlyClaire** [comment]
Fixed by #16270
